Data/Scripts/GardenConquest/Extensions/GridExtensions.cs

```diff
@@ -85,6 +85,54 @@ public static class GridExtensions {
 			return true;
 		}
 
+		/// <summary>
+		/// Is the given player allowed to see the given grid's position?
+		/// Player is able to see the position of the grid if the player
+		/// is part of the mainCockpit's faction OR the player owns the mainCockpit
+		/// </summary>
+		/// <param name="playerID"></param>
+		/// <param name="grid"></param>
+		/// <remarks>
+		/// Relatively expensive since we iterate through all blocks in grid!
+		/// </remarks>
+		/// <returns></returns>
+		public static bool canDisplayPositionTo(this IMyCubeGrid grid, long playerID) {
+			bool result = false;
+			List<IMySlimBlock> fatBlocks = new List<IMySlimBlock>();
+
+			// Get only FatBlocks from the blocks list from the grid
+			Func<IMySlimBlock, bool> isFatBlock = b => b.FatBlock != null;
```

Comment:
As we discussed, try to get some of the additional checks from lines 112 & 113 in this filter function.